primitives/rpc/src/number.rs

```diff
@@ -67,24 +67,27 @@ pub struct TryFromIntError(pub(crate) ());
 impl TryFrom<NumberOrHex> for u32 {
 	type Error = TryFromIntError;
 	fn try_from(num_or_hex: NumberOrHex) -> Result<u32, TryFromIntError> {
-		let num_or_hex = num_or_hex.into_u256();
-		if num_or_hex > U256::from(u32::max_value()) {
-			return Err(TryFromIntError(()));
-		} else {
-			Ok(num_or_hex.as_u32())
-		}
+		num_or_hex.into_u256().try_into().map_err(|_| TryFromIntError(()))
 	}
 }
 
 impl TryFrom<NumberOrHex> for u64 {
 	type Error = TryFromIntError;
 	fn try_from(num_or_hex: NumberOrHex) -> Result<u64, TryFromIntError> {
-		let num_or_hex = num_or_hex.into_u256();
-		if num_or_hex > U256::from(u64::max_value()) {
-			return Err(TryFromIntError(()));
-		} else {
-			Ok(num_or_hex.as_u64())
-		}
+		num_or_hex.into_u256().try_into().map_err(|_| TryFromIntError(()))
+	}
+}
+
+impl TryFrom<NumberOrHex> for u128 {
+	type Error = TryFromIntError;
+	fn try_from(num_or_hex: NumberOrHex) -> Result<u128, TryFromIntError> {
+		num_or_hex.into_u256().try_into().map_err(|_| TryFromIntError(()))
+	}
+}
+
+impl From<NumberOrHex> for U256 {
```

Comment:
Can we not just make on generalized version:

```
impl<T: TryFrom<U256>> From<NumberOrHex> for T {
}
```
?

Reply:
unfortunately it conflicts:
```
error[E0119]: conflicting implementations of trait `std::convert::TryFrom<number::NumberOrHex>`:
  --> primitives/rpc/src/number.rs:67:1
   |
67 | impl<T: TryFrom<U256>> TryFrom<NumberOrHex> for T {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T, U> TryFrom<U> for T
             where U: Into<T>;

```

but I can do a macro to implement to way more integer types

Reply:
Ahh, always this error. No, I think it is fine :)